cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -247,39 +252,26 @@ public String getHeadlessServiceName() {
      * @return The selector labels as an instance of the Labels object.
      */
     public Labels getSelectorLabels() {
-        return labels.withName(name).strimziSelectorLabels();
+        return getLabelsWithStrimziName(name, Collections.emptyMap()).strimziSelectorLabels();
     }
 
-    /**
-     * @return The selector labels as Map.
-     */
-    public Map<String, String> getSelectorLabelsAsMap() {
-        return getSelectorLabels().toMap();
+    protected Labels getLabelsWithStrimziName(String name, Map<String, String> additionalLabels) {
+        return labels.withStrimziName(name).withAdditionalLabels(additionalLabels);
     }
 
-    protected Map<String, String> getLabelsWithName() {
-        return getLabelsWithName(name);
+    protected Labels getLabelsWithNameAndDiscovery(String name, Map<String, String> additionalLabels) {
+        return getLabelsWithStrimziName(name, additionalLabels).withStrimziDiscovery();
     }
 
-    protected Map<String, String> getLabelsWithName(Map<String, String> userLabels) {
-        return getLabelsWithName(name, userLabels);
-    }
-
-    protected Map<String, String> getLabelsWithName(String name) {
-        return labels.withName(name).toMap();
-    }
-
-    protected Map<String, String> getLabelsWithName(String name, Map<String, String> userLabels) {
-        return labels.withName(name).withUserLabels(userLabels).toMap();
+    /**
+     * @param resource
+     * @return The default Labels set
+     */
+    protected Labels generateDefaultLabels(HasMetadata resource) {
```

Comment:
If I read it corrctly, this seems to be called only from one place. Shouldn't we use the `Labels.geenrateDefaultLabels()` directly and remove this?

Reply:
You're absolutely right I will change this, I thought changing the signature would be useful but it wasn't